### PR TITLE
Adds Emojis to the workload get output

### LINKED ISSUE #262: Decorate workload get output
### Description of problem

The workload get output has added a lot of valuable information, and while it is organized in chunks, they visually blend together. 

While indenting sections under headlines helps clarify the structure of the content, there is an opportunity to add more visual distinction and context.

### Proposed solution

EMOJIS!

Tiny images would add visual interest to a text-dominated interface
It would also strengthen visual hierarchy to improve scannability, call attention to important elements, and potentially contextualize what is happening in a give section.

```gherkin
Given I 'get' information about a workload
When the information is printed
Then each section headline is prefixed with an emoji.
```

### Example

```
$ tanzu apps workload get NAME

---
Name: tanzu-java-web-app: Ready
---

💾 Source  
   type:     git
   url:      https://github.com/spring-projects/spring-petclinic
   branch:   main

📦 Supply chain  
   name:          source-to-url

   RESOURCE           READY   HEALTHY   TIME       
   Source Provider    True    True      9m ago
   Image Builder      True    True      8m ago
   Config Provider    True    True      7m ago
   App Config         True    True      6m ago

🚚 Delivery  
   name:          source-to-url

   RESOURCE           READY   HEALTHY   TIME       
   Source Provider    True    True      9m ago
   Image Builder      True    True      8m ago
   Config Provider    True    True      7m ago
   App Config         True    True      6m ago

💬 Messages
   No Messages Found


🔎 Resources
   NAME                 STATUS   RUNTIME   URL
   tanzu-java-web-app   Ready    knative   http://tanzu-tap.example.com


🔁 Services
   CLAIM      NAME         KIND         API VERSION
   database   my-prod-db   PostgreSQL   1.1.1

```

### Describe alternatives you've considered

Source
```
🔢 - :1234: - save / code
💾 - floppy disk - save / file
👩🏽‍💻 - woman technologist - application developer
📥 - inbox tray - receiving / input
📜 - scroll - long document, ancient and inscrutable
📁 - folder - organizing / structure / file location
```

Supply Chain
```
🚚 - delivery truck - transport / infrastructure / resources
🔄 - counter clockwise arrows - continuous cycle - metaphor for reconciling
✉️ - envelope - delivery / message 
📦 - package - delivery / package
🗺 - map - plan / directions (cartography tie-in)
[] - factory - 
```

Delivery
```
🚚 - delivery truck - transport / infrastructure / resources
🚝 - train - infrastructure  🚄 🚉
🏁 - checkered flag - finish line 
```

Messages
```
🤖 - robot face - automated work
✉️  - envelope - delivery / message - renders small sometimes
📣 - megaphone - announce / loud
💬 - speech balloon - talking 
```

Resources
```
🔎 - magnifying glass - examination / investigation / details
🪵 - wood - basic resource, building material
🖥 - desktop computer - computer / infrastructure
🐢 - turtle - turtles all the way down
🪆 - nesting dolls - things inside things 
🗄 - file cabinet - organization / things in things
```

Services
```
🪢 - knot - tied / bound / connected / intertwined
🤖 - robot face - automated work
🥡 - takeout box - delivery / to-go
↔️ -  - interchange / back and forth
🔁 -  - interchange / back and forth
```

### PR DESCRIPTION
**Pull request by shaheerkootteeri**
Pull request

### What this PR does / why we need it

PR adds emojis to the workload get output command
### Which issue(s) this PR fixes

Fixes #262 

### Describe testing done for PR

Added unit test cases with Emojis. Modified integration test to validate the emojis in the output. tested in local cluster.

### CODE REVIEW

**Comment by shaheerkootteeri** [comment]
image

File: pkg/cli-runtime/config.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ import (
 
 const defaultTanzuIgnoreFile = ".tanzuignore"
 
+var space = " "
```
Comment (odinnordico) [inline pkg/cli-runtime/config.go:34]: there is no need  to use a spacer constant, just put the space in the format directly

File: pkg/cli-runtime/emojidata.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package cli
```
Comment (odinnordico) [inline pkg/cli-runtime/emojidata.go:1]: rename the file to `emoji.go`

File: pkg/cli-runtime/emojidata.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+type Icon rune
+
+const (
+	FloppyDisk    Icon = '💾'
```
Comment (odinnordico) [inline pkg/cli-runtime/emojidata.go:6]: package `cli` does not explain what the constants are, add a prefix `Icon` to each const i.e. `DeliveryIcon`

**Comment by shaheerkootteeri** [comment]
image

**Comment by shaheerkootteeri** [comment]
image

File: pkg/cli-runtime/emojidata.go (hunk 3 above)
Comment (shaheerkootteeri) [reply pkg/cli-runtime/emojidata.go]: These icons could be shared for different headers or messages, adding suffix could be misleading.

File: pkg/cli-runtime/config.go (hunk 1 above)
Comment (shaheerkootteeri) [reply pkg/cli-runtime/config.go]: Taken care

File: pkg/cli-runtime/emojidata.go (hunk 2 above)
Comment (shaheerkootteeri) [reply pkg/cli-runtime/emojidata.go]: Updated the file name

File: pkg/cli-runtime/config.go [hunk 4]
```diff
@@ -103,11 +106,14 @@ func (c *Config) Efaintf(format string, a ...interface{}) (n int, err error) {
...
 }
+func (c *Config) EmojiBoldf(emoji Icon, format string, a ...interface{}) (n int, err error) {
+	emojiFormat := fmt.Sprintf("%s%s%s", string(emoji), " ", format)
+	return printer.BoldColor.Fprintf(c.Stdout, emojiFormat, a...)
+}
 
 func (c *Config) Eboldf(format string, a ...interface{}) (n int, err error) {
 	return printer.BoldColor.Fprintf(c.Stderr, format, a...)
 }
-
```
Comment (shashwathi) [inline pkg/cli-runtime/config.go:110]: do not delete this line

File: testing/e2e/workload_test.go [hunk 5]
```diff
@@ -298,8 +302,19 @@ func TestCreateFromGitWithAnnotations(t *testing.T) {
...
 			WorkloadName: "test-create-git-annotations-workload",
-			Command: *it.NewTanzuAppsCommandLine(
-				"workload", "get", "test-create-git-annotations-workload", namespaceFlag),
+			Command: func() it.CommandLine {
+				c := *it.NewTanzuAppsCommandLine(
+					"workload", "get", "test-create-git-annotations-workload", namespaceFlag)
+				return c
+			}(),
+			Verify: func(t *testing.T, output string, err error) {
+				decodedString := strconv.QuoteToASCII(output)
```
Comment (shashwathi) [inline testing/e2e/workload_test.go:311]: this is awesome

File: testing/e2e/workload_test.go [hunk 6]
```diff
@@ -42,6 +44,8 @@ var (
 	}
 )
 
+const pattern = "\\\\U000[a-zA-Z0-9]+"
```
Comment (shashwathi) [inline testing/e2e/workload_test.go:47]:
```suggestion
var (
pattern = "\\\\U000[a-zA-Z0-9]+"
regex = regexp.MustCompile(pattern)
)
```

File: testing/e2e/workload_test.go [hunk 7]
```diff
@@ -316,3 +331,8 @@ func TestCreateFromGitWithAnnotations(t *testing.T) {
 	}
 	testSuite.Run(t)
 }
+func foundMatch(pattern string, testCase string) bool {
+	regex := regexp.MustCompile(pattern)
```
Comment (shashwathi) [inline testing/e2e/workload_test.go:335]: delete `foundMatch` func and call `FindString` inside the test case

**Review by shashwathi — CHANGES_REQUESTED** [review]

File: testing/e2e/workload_test.go (hunk 7 above)
Comment (shaheerkootteeri) [reply testing/e2e/workload_test.go]: Took care in the latest commit

File: testing/e2e/workload_test.go (hunk 6 above)
Comment (shaheerkootteeri) [reply testing/e2e/workload_test.go]: Updated the code

File: testing/e2e/workload_test.go (hunk 5 above)
Comment (shaheerkootteeri) [reply testing/e2e/workload_test.go:314]: Thank you

File: pkg/cli-runtime/config.go (hunk 4 above)
Comment (shaheerkootteeri) [reply pkg/cli-runtime/config.go]: Updated the code

File: pkg/cli-runtime/config_test.go [hunk 8]
```diff
@@ -153,3 +153,47 @@ func TestConfig_Print(t *testing.T) {
 		})
 	}
 }
+func TestConfig_EmojiPrint(t *testing.T) {
```
Comment (shashwathi) [inline pkg/cli-runtime/config_test.go:156]: You can merge these 2 tests into 1. You can set the printer `EmojiBoldf` or `EmojiSuccessf` in the test case. You can avoid the duplication

Comment (shaheerkootteeri) [reply pkg/cli-runtime/config_test.go:156]: Done

File: pkg/cli-runtime/config_test.go [hunk 9]
```diff
@@ -153,3 +153,51 @@ func TestConfig_Print(t *testing.T) {
...
+	scheme := runtime.NewScheme()
+	config := cli.NewDefaultConfig("test", scheme)
+
+	tests := []struct {
+		name    string
+		format  string
```
Comment (shashwathi) [inline pkg/cli-runtime/config_test.go:162]: remove format field

File: pkg/cli-runtime/config_test.go [hunk 10]
```diff
@@ -153,3 +153,51 @@ func TestConfig_Print(t *testing.T) {
...
+		icon    cli.Icon
+		args    []interface{}
+		printer func(icon cli.Icon, format string, a ...interface{}) (n int, err error)
+		stdout  string
+		stderr  string
+	}{{
+		name:    "EmojiSuccessf",
+		format:  "%s",
+		icon:    cli.FloppyDisk,
+		args:    []interface{}{"Source"},
```
Comment (shashwathi) [inline pkg/cli-runtime/config_test.go:172]: update args to some other variable in one of the tests

**Review by shashwathi — COMMENTED** [review]
Resolve conflicts and couple of minor comments

File: pkg/cli-runtime/config_test.go (hunk 10 above)
Comment (shaheerkootteeri) [reply pkg/cli-runtime/config_test.go]: Done

**Review by shashwathi — APPROVED** [review]

File: pkg/cli-runtime/emoji.go [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+	Maginifying     Icon = '🔎'
+	Repeat          Icon = '🔁'
+	Antenna         Icon = '📡'
+	Ship            Icon = '🚢'
+	Envelop         Icon = '✉'
+	IncomingEnvelop Icon = '📨'
+	Canoe           Icon = '🛶'
+	Inbox           Icon = '📥'
+	Question        Icon = '❓'
+	ThumpsUp        Icon = '👍'
```
Comment (warango4) [inline pkg/cli-runtime/emoji.go:35]: Typo:
Should be: ThumbsUp

**Pull request merged**